### PR TITLE
otb-tracker: Dynamic timeout

### PR DESCRIPTION
**Pull request by Tikiwinkie**
Rebase with PR: Fix get-ip-asn

### CODE REVIEW

**Comment by gregdel** [comment]
> 
>   (2.2) When the first RTT measurement R is made, the host MUST set
>
>          SRTT <- R
>            RTTVAR <- R/2
>            RTO <- SRTT + max (G, K*RTTVAR)
>
>         where K = 4.

But [K = 3 here](https://github.com/ovh/overthebox-feeds/pull/644/files#diff-c4442a551dca8a57118552973e38aee3R28), can you comment on the things you've changed from the RFC and explain why ?

File: otb-tracker/bin [hunk 1]
```diff
@@ -16,39 +16,99 @@ export OTB_TRACKER_INTERFACE
...
 export OTB_TRACKER_RESULT
 export OTB_TRACKER_DEVICE
-export OTB_TRACKER_INTERFACE_IP
-export OTB_TRACKER_INTERFACE_GATEWAY
+export OTB_TRACKER_DEVICE_IP
+export OTB_TRACKER_DEVICE_GATEWAY
+
+# set constants for rto updating
+# we've changed the default values of the RFC to K=3 and beta=0.25 instead of K=4 and beta=0.125
+# this allows us to lower the gap between rtt and rto when the latency increases quickly as our measurements are less frequent than the standard timeout computation of the TCP stack
```
Comment (gregdel) [inline otb-tracker/bin]: That's a looooooooooooooooooooooooooonnnnnnnnnnnngggggggggggggg line !
You should wrap this up

File: otb-tracker/bin [hunk 2]
```diff
@@ -16,39 +16,99 @@ export OTB_TRACKER_INTERFACE
...
-	OTB_TRACKER_INTERFACE_IP=
-	OTB_TRACKER_INTERFACE_GATEWAY=
+	# setup tracker variables
+	OTB_TRACKER_DEVICE_IP=
+	OTB_TRACKER_DEVICE_GATEWAY=
 	OTB_TRACKER_DEVICE=
 	OTB_TRACKER_STATUS="ERROR"
 	OTB_TRACKER_RESULT=
+	_update_timeout
+
```
Comment (gregdel) [inline otb-tracker/bin]: useless line ?

**Comment by Tikiwinkie** [comment]
Changes done

File: otb/Makefile [hunk 3]
```diff
@@ -9,7 +9,7 @@ include $(INCLUDE_DIR)/package.mk
 MY_DEPENDS := \
     otb-backup otb-diagnostics \
     otb-qos otb-dscp otb-remote \
-    otb-tracker otb-autoroute \
```
Comment (gregdel) [inline otb/Makefile]: I don't think we should change this..

File: overthebox/Makefile [hunk 4]
```diff
@@ -8,7 +8,7 @@ include $(INCLUDE_DIR)/package.mk
 
 MY_DEPENDS := \
     graph glorytun glorytun-udp mptcp shadowsocks-libev  \
-    jq curl ca-bundle ca-certificates
+    jq curl ca-bundle ca-certificates otb-tracker
```
Comment (gregdel) [inline overthebox/Makefile]: Why ?

File: otb-tracker/init [hunk 5]
```diff
@@ -44,8 +45,9 @@ _launch_tracker() {
 	procd_set_param command /bin/otb-tracker "$1" "$call" $options
 	procd_append_param env OTB_TRACKER_HOST="$host"
 	procd_append_param env OTB_TRACKER_TIMEOUT="$timeout"
-	procd_append_param env OTB_TRACKER_RETRY="$retry"
+	procd_append_param env OTB_TRACKER_TRIES="$tries"
 	procd_append_param env OTB_TRACKER_INTERVAL="$interval"
+	procd_append_param env OTB_TRACKER_POST_TRACKING_TIMEOUT="$post_tracking_timeout"
```
Comment (gregdel) [inline otb-tracker/init]:
```
root@OverTheBox:/# /etc/init.d/otb-tracker start
/etc/rc.common: eval: line 1: options:string:tracker.overthebox.ovh: not found
```

File: otb-tracker/init [hunk 6]
```diff
@@ -11,8 +11,9 @@ _validate_section() {
 		'call:string:dns' \
 		'host:host:51.254.49.132' \
 		'timeout:uinteger:3' \
-		'retry:uinteger:2' \
+		'tries:uinteger:3' \
 		'interval:uinteger:2' \
+		'post_tracking_timeout:uinteger:2'
 		'options:string:tracker.overthebox.ovh'
 }
```
Comment (gregdel) [inline otb-tracker/init]:
```
root@OverTheBox:/# /etc/init.d/otb-tracker start
/etc/rc.common: eval: line 1: options:string:tracker.overthebox.ovh: not found
```

File: otb-tracker/bin [hunk 7]
```diff
@@ -16,39 +16,100 @@ export OTB_TRACKER_INTERFACE
...
+config_get OTB_TRACKER_DEVICE "$OTB_TRACKER_INTERFACE" ifname
 
 # main loop
 while true; do
-	OTB_TRACKER_INTERFACE_IP=
-	OTB_TRACKER_INTERFACE_GATEWAY=
+	# setup tracker variables
+	OTB_TRACKER_DEVICE_IP=
+	OTB_TRACKER_DEVICE_GATEWAY=
 	OTB_TRACKER_DEVICE=
```
Comment (gregdel) [inline otb-tracker/bin]: Mmmm does this reset the value we previously set line 70 ?

**Comment by gregdel** [comment]
Ping @angt for post-review

**Pull request merged**